### PR TITLE
feat: add metadata update notifier

### PR DESCRIPTION
**Pull request by whylee259**

### CODE REVIEW

File: contracts/token/ERC1155/MultiToken.sol [hunk 1]
```diff
@@ -125,6 +152,14 @@ contract MultiToken is
         emit MintApproval(minter, approved);
     }
 
+    function setMetadataOperatorPermission(address operator, bool grant)
+        external
+        onlyOwner
+    {
+        metadataOperators[operator] = grant;
```
Comment (jyeoniii) [inline contracts/token/ERC1155/MultiToken.sol:163]: nit: how about checking `require(metadataOperators[operator] != grant)`?

File: contracts/token/ERC721/ItemNFT.sol [hunk 2]
```diff
@@ -112,11 +133,20 @@ contract ItemNFT is
...
     }
 
-    function setTokenURI(uint256 tokenId, string calldata _tokenURI)
-        public
+    function setMetadataOperatorPermission(address operator, bool grant)
+        external
         onlyOwner
     {
-        _setTokenURI(tokenId, _tokenURI);
+        metadataOperators[operator] = grant;
```
Comment (jyeoniii) [inline contracts/token/ERC721/ItemNFT.sol:144]: ditto

File: contracts/token/ERC721/ItemNFTSnapshot.sol [hunk 3]
```diff
@@ -126,11 +144,20 @@ contract ItemNFTSnapshot is
...
     }
 
-    function setTokenURI(uint256 tokenId, string calldata _tokenURI)
-        public
+    function setMetadataOperatorPermission(address operator, bool grant)
+        external
         onlyOwner
     {
-        _setTokenURI(tokenId, _tokenURI);
+        metadataOperators[operator] = grant;
```
Comment (jyeoniii) [inline contracts/token/ERC721/ItemNFTSnapshot.sol:155]: ditto

**Review by jyeoniii — APPROVED** [review]
Other than minor comments, LGTM

File: contracts/token/ERC1155/MultiToken.sol (hunk 1 above)
Comment (whylee259) [reply contracts/token/ERC1155/MultiToken.sol:163]: Applied. thx!

File: contracts/token/ERC721/ItemNFTSnapshot.sol (hunk 3 above)
Comment (whylee259) [reply contracts/token/ERC721/ItemNFTSnapshot.sol:155]: Applied. thx!

File: contracts/token/ERC721/ItemNFT.sol (hunk 2 above)
Comment (whylee259) [reply contracts/token/ERC721/ItemNFT.sol:144]: Applied. thx!

**Pull request merged**